### ISSUE TITLE
(code=exited, status=126)

### ISSUE THREAD

**Issue by renanagg** [abcd567a/mm2#7]
Hi, I'm having a problem installing MM2 and I can't find the solution.
Here is the message it is giving:

● mm2.service - ModeSMixer2
   Loaded: loaded (/lib/systemd/system/mm2.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Fri 2023-10-06 03:05:58 BST; 9s ago
  Process: 5296 ExecStart=/bin/bash /usr/share/mm2/mm2.sh (code=exited, status=126)
 Main PID: 5296 (code=exited, status=126)

**Comment by abcd567a** [comment]
(1) On which OS you have installed mm2?

(2) what is output of following command?

`sudo modesmixer2  `

**Comment by Maillouski** [comment]
Hi, I also have an issue, here's some information.

RPi 4 running bookworm

/usr/share/mm2/modesmixer2: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory

the default version of libssl for RpiOS bookworm (debian based) is libssl 3.0 and modesmixer2 wants libssl 1.1 and libcrypto 1.1

**Comment by abcd567a** [comment]
Yes, the last version released by developer of modesmixer2 was for Debian 10 (buster) . It worked ok on Debian 11 (bullseye), but failed on Debian 12 (bookworm) because version of libssl and libcrypto in bookworm are higger than required by modesmixer2.

Unfortunately the developer Sergsero has about 2 years ago abandoned this project and has pulled down his site xdeco2. No chance now to get a version compatible to bookworm.
In view of this situation, I tried to "assemble" a mixer out of dump1090-mutability / dump1090-fa and used socat to creat connections.  The work is still under development.

### (1) https://github.com/abcd567a/mixer/blob/master/README.md

### (2) [Alternate to ModeSMixer2 on Bookworm](https://discussions.flightaware.com/t/alternate-to-modesmixer2-on-bookworm/89786)

**Comment by renanagg** [comment]
Good morning!
I understand, it's a shame he abandoned this project.

That's great that you're trying to find alternatives. Taking this
opportunity, I want to thank you very much for your valuable help in
providing guidance, tips and support to the community. I've lost count of
how many times you've helped me with guidance spread across different
forums!

Thank you for your response and congratulations on your willingness to
always help us!

Hugs!

Em qua., 18 de out. de 2023 às 20:46, abcd567a ***@***.***>
escreveu:

**Comment by zoliqe** [comment]
try run `sudo apt install libssl1.1` this helped me on rasbian os (which is based on bookworm).

**Comment by abcd567a** [comment]
> try run `sudo apt install libssl1.1` this helped me on rasbian os (which is based on bookworm).

Thank you very much for providing this solution. 
I have now added  `apt install libssl1.1`  in the installation scripts.
I have also added `apt install libstdc++6` in installation scripts, as this was also required on some distros.
